platform/security/interceptor/security-interceptor-guest/src/main/java/org/codice/ddf/security/interceptor/GuestInterceptorSansPolicy.java

```diff
@@ -0,0 +1,134 @@
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.xml.soap.SOAPElement;
+import javax.xml.soap.SOAPException;
+import javax.xml.soap.SOAPFactory;
+
+import org.apache.cxf.binding.soap.SoapMessage;
+import org.apache.cxf.interceptor.Fault;
+import org.apache.cxf.interceptor.security.DefaultSecurityContext;
+import org.apache.cxf.phase.Phase;
+import org.apache.cxf.security.SecurityContext;
+import org.apache.cxf.transport.http.AbstractHTTPDestination;
+import org.apache.cxf.ws.security.tokenstore.SecurityToken;
+import org.apache.cxf.ws.security.wss4j.AbstractWSS4JInterceptor;
+import org.apache.cxf.ws.security.wss4j.PolicyBasedWSS4JInInterceptor;
+import org.apache.cxf.ws.security.wss4j.WSS4JInInterceptor;
+import org.apache.shiro.subject.PrincipalCollection;
+import org.apache.wss4j.common.ext.WSSecurityException;
+import org.apache.wss4j.common.principal.SAMLTokenPrincipalImpl;
+import org.apache.wss4j.common.saml.SamlAssertionWrapper;
+import org.codice.ddf.security.handler.api.BaseAuthenticationToken;
+import org.codice.ddf.security.handler.api.GuestAuthenticationToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Element;
+
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.common.util.Security;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+
+/**
+ * Interceptor for guest access to SOAP endpoints.
+ */
+public class GuestInterceptorSansPolicy extends AbstractWSS4JInterceptor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GuestInterceptorSansPolicy.class);
+
+    private SecurityManager securityManager;
+
+    private Map<String, Subject> cachedGuestSubjectMap = new HashMap<>();
+
+    public GuestInterceptorSansPolicy(SecurityManager securityManager) {
+        super();
+        LOGGER.trace("Constructing Guest Interceptor.");
+        this.securityManager = securityManager;
+        setPhase(Phase.PRE_PROTOCOL);
+        //make sure this interceptor runs before the WSS4J one in the same Phase, otherwise it won't work
+        Set<String> before = getBefore();
+        before.add(WSS4JInInterceptor.class.getName());
+        before.add(PolicyBasedWSS4JInInterceptor.class.getName());
+        LOGGER.trace("Exiting Guest Interceptor constructor.");
+    }
+
+    @Override
+    public void handleMessage(SoapMessage message) throws Fault {
+
```

Comment:
There's no way to turn off guest access here. Is that what we want? That definitely seems wrong.